docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
-}
-
-```
+#### Transport Adapter Structure
 
-### Initialization
+|||
+Figure 3: Transport Adapter UML Diagram
+![TA](./assets/ta.png)
+|||
 
-By default, the TA is initialized during TM initialization. If the TA needs to be added to a TM which has already been initialized, the TA should be initialized before it is added the TM
+### Transport Adapter Workers
 
-!!! note
+The Device Scanner is responsible for scanning for new devices to connect with. When a device is found, this worker is responsible for alerting the Transport Adapter, as well as alerting the Transport Manager via the Transport Adapter Listener. Next, the Transport Manager will instruct the Transport Adapter to connect with the devices.
 
-The TM has a TCP adapter by default. Adding the new instance of the same Adapter may result in unexpected behavior. The developer must change the standard behavior of the TCP adapter to elinatate any potential problems.
+The Client Connection Listener implements receiving a connection that is originated by a device. This will typically wait for connection from a device, then establish that connection, finally alerting the Transport Manager via the Transport Manager Listener of the newly connected device and app IDs.
 
-!!!
+The Server Connection Factory implements a connection that is originated from Core. For example, Core reaches out to a predefined web address to start a cloud application. This type of communication requires that the Transport Adapter knows of the device and application in advance. When this connection is created, the Transport Adapter will alert the Transport Manager of the new devices and applications in a similar fashion to other workers.
 
-### Custom Transport Adapter Implementations
+Depending on what your type of transport is, whether Core will be the server or the client, you will likely implement either the Client Connection Listener or the Server Connection Factory.
 
-If the default implementation of TAs is not suitable, you can create a transport adapter from scratch.
+#### Client Connection Listener
 
-  * Create a custom class that implements the transport adapter interface. The custom TA should use the defined interface TransportAdapterListener for notifying the Transport Manager
-  * The instance of TA listener will be set by the TM automatically when a custom TA is added. If the custom TA does not store its listener then the adapter will not be able to notify the TM about events such as OnDataReceiveDone or OnConnectDone
+Since the TCP Transport Adapter is running a server that clients will connect to, much of the implementation lives in the Client Connection Listener. The Client Connection Listener has a fair few functions to implement, let’s start with Init(). 
 
 ```
-#include "transport_adapter.h"
+// tcp_client_listener.cc
+// Code has been simplified and whitespace has been added for readability.
+// Thank you to Sho Amano for the helpful comments.
+
+TransportAdapter::Error TcpClientListener::Init() {
```

Comment:
I think it would be helpful to link to where this code comes from on github

Reply:
I am not sure it would be super helpful for a couple reasons:
- we can only link to a line (more maintaining)
- this code looks very different than what is actually on github (might be more confusing)
- if someone is reading this guide, they can probably find this code pretty quick, right?

Let me know if you still think I should add it...

Reply:
Good points.. Maybe the better solution would be to link to some of the important transport classes at the end of the guide?

Reply:
I added a few links throughout, everywhere I mention a specific component. Lmk where you think more would help